File: networkmodule/src/main/java/com/test/networkmodule/di/NetworkComponent.java
package com.test.networkmodule.di;

import com.test.networkmodule.NetworkSDK;
import com.test.networkmodule.di.network.CommonNetworkModule;

import javax.inject.Singleton;

import dagger.Component;


@Singleton
@Component(modules = {
        CommonNetworkModule.class,
        ShoppingHostingModule.class})
public interface NetworkComponent {
    void inject(NetworkSDK networkSDK);

    @Component.Builder
    interface Builder {

        NetworkComponent build();
    }
}

File: networkmodule/src/main/java/com/test/networkmodule/di/NetworkComponent.kt
package com.test.networkmodule.di

import com.test.networkmodule.NetworkSDK
import com.test.networkmodule.di.network.CommonNetworkModule
import dagger.Component
import javax.inject.Singleton


@Singleton
@Component(modules = [CommonNetworkModule::class, ShoppingHostingModule::class])
interface NetworkComponent {
    fun inject(networkSDK: NetworkSDK)

    @Component.Builder
    interface Builder {

        fun build(): NetworkComponent
    }
}
